rpc/server.go

```diff
@@ -13,31 +12,49 @@ import (
 
 type rpcServer struct {
 	bus modules.Bus
+
+	logger modules.Logger
 }
 
-var _ ServerInterface = &rpcServer{}
-var _ modules.IntegratableModule = &rpcServer{}
+var (
+	_ ServerInterface            = &rpcServer{}
+	_ modules.IntegratableModule = &rpcServer{}
+)
 
 func NewRPCServer(bus modules.Bus) *rpcServer {
 	s := &rpcServer{}
 	s.SetBus(bus)
+
 	return s
 }
 
-func (s *rpcServer) StartRPC(port string, timeout uint64) {
-	log.Printf("Starting RPC on port %s...\n", port)
+func (s *rpcServer) StartRPC(port string, timeout uint64, logger modules.Logger) {
+	s.logger = logger
+
+	s.logger.Info().Msgf("Starting RPC on port " + port)
 
 	e := echo.New()
 	middlewares := []echo.MiddlewareFunc{
-		middleware.Logger(),
+		middleware.RequestLoggerWithConfig(middleware.RequestLoggerConfig{
+			LogURI:    true,
+			LogStatus: true,
+			LogValuesFunc: func(c echo.Context, v middleware.RequestLoggerValues) error {
```

Comment:
I understand why you were asking about the lack of a `Context` now. 

Like I said, we used to have a `PocketContext` that would embed a `golang Context` inside of it, but it ended up being overkill/unnecessary for what we have.

Keep this in mind for the future, as I can see us bringing it when if/when the time is right.